src/olympia/devhub/tasks.py

```diff
@@ -414,20 +415,44 @@ def run_addons_linter(path, listed=True):
     if not listed:
         args.append('--self-hosted')
 
+    stdout, stderr = tempfile.TemporaryFile(), tempfile.TemporaryFile()
+
     process = subprocess.Popen(
-        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        args,
+        stdout=stdout,
+        stderr=stderr,
+        # default but explicitly set to make sure we don't open a shell.
+        shell=False,
+        # Avoid buffering to make sure we don't accidently trim some output.
+        bufsize=0
```

Comment:
Isn't that plus using a file going to slow it down considerably for every validation ?